### ISSUE TITLE
panic: runtime error: index out of range on main.GetSNBlock

### ISSUE THREAD

**Issue by TimDorand** [jdavanne/docker-sni-proxy#2]
Docker exits on runtime error after 
``` 
level=error msg="Session 1863 - Not SSL Hello : Finished parsing the Extension block without finding an SN block
```

Full log:

```
time="2018-08-07 23:59:38.510624142" level=error msg="Session 1863 - Not SSL Hello : Finished parsing the Extension block without finding an SN block "[j2j[��S/F�G�������r9J2=I�������0�,�(�$��
�kj98���2�.�*�&���=5���
�/�+�'�#��	��g@32��ED�1�-�)�%���</�A���
                                              �	�m
                                                           
2

 	
 "
time="2018-08-08 00:19:41.911881868" level=info msg="Session 1864 - Incoming connection from  54.198.179.231:42685"
time="2018-08-08 00:19:41.914321338" level=info msg="Session 1864 - Hello TLS Packet, version: 1.3 size: 517 517"
panic: runtime error: index out of range

goroutine 9851 [running]:
main.GetSNBlock(0xc4200fe0b2, 0x153, 0x41b6, 0xc4200fe0b2, 0x153, 0x41b6, 0x0, 0x0)
	/app/src/src/tls-hello-sni.go:93 +0x270
main.GetHostname(0xc4200fe000, 0x205, 0x4268, 0xc4201c6020, 0x2, 0xc4201c6008, 0x3)
	/app/src/src/tls-hello-sni.go:44 +0xec
main.(*Session).ServerDispatch(0xc4202e2040, 0x8c92c0, 0xc420334000, 0x1, 0xc420018005, 0x7)
	/app/src/src/session.go:95 +0x4cf
created by main.serve
	/app/src/src/main.go:31 +0x1a0
```

**Comment by TimDorand** [comment]
Fixed in the commit 5f25d418c3023e100cd2f5a258ef6b5aee90167d

**Comment by TimDorand** [comment]
Fixed in the commit 5f25d418c3023e100cd2f5a258ef6b5aee90167d
